### ISSUE TITLE
Potential logic issue in sem_criterion label assignment for clean samples during refinement phase

### ISSUE THREAD

**Issue by XinhanWang** [yangaonidaye123/NRCL#7]
**Description**
Hi, thank you for your great work on NRCL. While reviewing the code in the refinement functions (e.g., `refinement_datasets` and `refinement_imgtxt`), I noticed a potential logic discrepancy regarding how labels are assigned to clean samples when calculating the semantic center loss (`sem_criterion`).

**Location**
In the refinement stage (around the `if args.yrectified:` block), the code defines labels for supervision as follows:

```python
if args.yrectified:
    y_final_one_hot = y_hat.clone().detach()
    y_final_one_hot[mask_clean,:] = y_clean # Correct: clean samples use GT one-hot
    
    # Potential Issue Here:
    y_final = torch.argmax(y_hat, dim=1)    # All samples use model predictions (y_hat)
    w = None
```

Later, when calculating the semantic loss for clean samples:
```python
y_final_s = y_final[mask_clean]
sem_loss_c, centers = sem_criterion(..., torch.cat((y_final_s, y_final_s), dim = 0), epoch)
```

**The Problem**
Since `y_final` is derived from `torch.argmax(y_hat, dim=1)`, the `y_final_s` for clean samples represents the **model's pseudo-label prediction** rather than the **ground truth label** (`target_c`). 

Even though these samples are confirmed as "clean" by the dataloader/noise-cleansing stage, they are being supervised by the model's own (possibly incorrect) predictions in the `sem_criterion`. This might lead to:
1. Suboptimal semantic center updates.
2. Slower convergence or error propagation for clean samples.

**Suggested Fix**
Should `y_final` for clean samples be forced to use the ground truth? For example:
```python
if args.yrectified:
    y_final_one_hot = y_hat.clone().detach()
    y_final_one_hot[mask_clean,:] = y_clean
    
    y_final = torch.argmax(y_hat, dim=1)
    # Suggested change:
    y_final[mask_clean] = target[mask_clean] 
```

I would appreciate it if you could clarify whether using model predictions for clean samples in `sem_criterion` was an intentional design choice or a minor oversight.

Best regards!
